runtime/src/main/java/org/corfudb/runtime/view/stream/AddressMapStreamView.java

```diff
@@ -400,7 +400,7 @@ protected long getMaxGlobalFromMaxEntries(int maxEntries) {
             // but we need to consider the case when the 'syncUpTo' address falls below the trim mark, so we correctly
             // report the TrimmedException allowing consumers to know that data is no longer available from this point.
             // We could throw the TrimmedException directly or in this case let remainingUpTo discover it for us.
-            if (getCurrentGlobalPosition() <= streamAddressSpace.getTrimMark()) {
+            if (getCurrentGlobalPosition() < streamAddressSpace.getTrimMark()) {
```

Comment:
Tried to understand the equal case but failed. Could you please explain a little more here.

Reply:
Sure, so the equal case is only valid on initial subscription, i.e., the exact moment when you do .subscribeListener(ts); For example, if someone requests to stream from the exact address that corresponds to the stream's trim mark, you are sure that address is no longer available and you can throw a TrimmedException (that is why the validateSyncPoint was added right on the subscribe method). 

But if you were allowed to subscribe and polling tasks are issued every Xms, you could reach a valid point where currentGlobalPosition = streamTrimMark, e.g., you are up to address = 10 (nothing has been trimmed)... you keep polling every Xms but no new updates come in (this happens for long time), then the log is trimmed (let's say at address 200), so now the streamTrimMark = 10... in this case they're equal but you didn't miss it, you had already processed 10 (but there were no further updates). 

For this reason we separated the case where you need the "=" on initial subscription (and moved that logic as a pre-validation as soon as you subscribe) and here removed it, as the pointer being at that position means it has already been resolved.